075-auto-rebalancing-cluster-scaling.md

```diff
@@ -0,0 +1,490 @@
+The "actual" `KafkaRebalance` custom resource will be named as `<my-cluster-name>-auto-rebalancing-<mode>` where the `<my-cluster-name>` part comes from the `metadata.name` in the `Kafka` custom resource, and the `<mode>` is referring to which rebalancing has to run (add or remove brokers).
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaRebalance
+metadata:
+  name: my-cluster-auto-rebalancing-add-brokers
+  finalizers:
+    - strimzi.io/auto-rebalancing
+spec:
+  mode: add-brokers
+  brokers: [3,4]
+  goals:
+    - CpuCapacityGoal
+    - NetworkInboundCapacityGoal
+    - DiskCapacityGoal
+    - RackAwareGoal
+    - MinTopicLeadersPerBrokerGoal
+    - NetworkOutboundCapacityGoal
+    - ReplicaCapacityGoal
+  skipHardGoalCheck: true
+  # ... other rebalancing related configuration
+```
+
+The operator also sets a finalizer, named `strimzi.io/auto-rebalancing`, on the "actual" `KafkaRebalance` custom resource.
+This is needed to avoid the user, or any other tooling, to delete the resource while the auto-rebalancing is still running.
+The finalizer is removed at the end of the auto-rebalancing process, with or without errors, allowing the "actual" `KafkaRebalance` custom resource deletion by the operator itself or by the user for manual intervention (more details later).
+
+The user has the flexibility to specify the [goals and the other rebalancing related configurations](https://strimzi.io/docs/operators/latest/configuring.html#type-KafkaRebalance-reference) the same way they already do today on a `KafkaRebalance` custom resource.
+This allows the user to not rely on the default Cruise Control goals configuration which would be common for any rebalance without overriding goals.
+They could define different goals based on the status of the Kafka cluster or skip hard goals in some scenarios for example.
+As already mentioned, specifying goals but more in general customizing the optimization proposal request is already possible in the current usage of a `KafkaRebalance` custom resource so it should be allowed for the auto-rebalancing as well.
+
+The idea of using a "template" `KafkaRebalance` custom resource could be also extended to the manual rebalancing operations started by the user when they create a new `KafkaRebalance` custom resource.
+It would mean a change on the `KafkaRebalance` custom resource by adding a new `spec.config` field to reference the rebalancing "template" and deprecating all the other fields used today (i.e. `spec.goals`, ...).
+Another option could be leaving both the ways living together.
+If this proposal is considered valid, doing such a change on the `KafkaRebalance` custom resource would help the proposal itself because the operator should not copy over the entire configuration from the "template" resource to the actual rebalance resource but just set the configuration reference.
+Of course, such a change would need a different proposal to be approved before this one.
```

Comment:
This was an idea which I put here to understand how much it was of interest but it would need a different proposal. We could even decide to not have it now but later on. Not sure why a different paragraph is needed.